nodejs/eks/cluster.ts

```diff
@@ -223,14 +225,38 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("instanceRole and instanceRoles are mutually exclusive, and cannot both be set.");
     }
 
-    // If no VPC was specified, use the default VPC.
+    if (args.subnetIds && (args.publicSubnetIds || args.privateSubnetIds)) {
+        throw new Error("subnetIds, and the use of publicSubnetIds and/or privateSubnetIds are mutually exclusive. Choose a single approach.");
+    }
+
+    // Configure default networking architecture.
     let vpcId: pulumi.Input<string> = args.vpcId!;
     let subnetIds: pulumi.Input<pulumi.Input<string>[]> = args.subnetIds!;
-    if (args.vpcId === undefined) {
+    const publicSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.publicSubnetIds!;
+    const privateSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.privateSubnetIds!;
+    let clusterSubnetIds: pulumi.Input<pulumi.Input<string>[]> = [];
+
+    // If no VPC is set, use the default VPC's subnets.
+    if (!args.vpcId) {
         const invokeOpts = { parent: parent };
         const vpc = aws.ec2.getVpc({ default: true }, invokeOpts);
         vpcId = vpc.then(v => v.id);
         subnetIds = vpc.then(v => aws.ec2.getSubnetIds({ vpcId: v.id }, invokeOpts)).then(subnets => subnets.ids);
+        clusterSubnetIds = subnetIds;
```

Comment:
There still seems to be something unnecessarily convoluted going on here. 

At the very least - this line is unnecessary. 

But it also still seems like in fact the entire use of subnetIds is unnecessary. Like if you just removed it and then inlined all the places that use it - everything would be the same but simpler. 

In the note above you point to code in a different file that uses core.subnetIds, but I’m talking about the subnetIds local variable here which is unrelated.

Reply:
I've removed the use of the local `subnetIds` variable, and inlined it. `subnetIds` only gets used in the following places:

- https://github.com/pulumi/pulumi-eks/blob/a869af60fc1a61049add65d622dc3202abfedd79/nodejs/eks/cluster.ts#L251
- https://github.com/pulumi/pulumi-eks/blob/a869af60fc1a61049add65d622dc3202abfedd79/nodejs/eks/cluster.ts#L474